.github/workflows/minimum_dependency_checker.yaml

```diff
@@ -0,0 +1,64 @@
+name: Minimum dependency checker
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'pyproject.toml'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Run min dep generator - test requirements
+        id: min_dep_gen_test
+        uses: alteryx/minimum-dependency-generator@v3.4
+        with:
+          paths: 'pyproject.toml'
+          options: 'dependencies'
+          extras_require: 'test'
+      - name: Save min test requirements and run diff
+        id: check_min_test
+        continue-on-error: true
+        run: |
+          mkdir /tmp/dependencies_updated_artifacts
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > /tmp/minimum_test_requirements.txt
+          diff /tmp/minimum_test_requirements.txt tests/dependency_update_check/minimum_test_requirements.txt
+      - name: if min test requirements have changed, write output file
+        if: steps.check_min_test.outcome != 'success'
+        run: |
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > tests/dependency_update_check/minimum_test_requirements.txt
+      - name: Run min dep generator - requirements
+        id: min_dep_gen_reqs
+        uses: alteryx/minimum-dependency-generator@v3.4
+        with:
+          paths: 'pyproject.toml'
+          options: 'dependencies'
+      - name: Save min requirements and run diff
+        id: check_min_reqs
+        continue-on-error: true
+        run: |
+          printf "${{ steps.min_dep_gen_reqs.outputs.min_reqs }}" > /tmp/minimum_requirements.txt
+          diff /tmp/minimum_requirements.txt tests/dependency_update_check/minimum_requirements.txt
+      - name: if min requirements have changed, write output file
+        if: steps.check_min_reqs.outcome != 'success'
+        run: |
+          printf "${{ steps.min_dep_gen_reqs.outputs.min_reqs }}" > tests/dependency_update_check/minimum_requirements.txt
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.MACHINEFL_DEPENDENCY_CHECKER_TOKEN }}
+          commit-message: Update minimum dependencies
+          title: Automated Minimum Dependency Updates
+          author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
+          body: "This is an auto-generated PR with **minimum** dependency updates.
+                 Please do not delete the `min-dep-update` branch because it's needed by the auto-dependency bot."
+          branch: min-dep-update
+          branch-suffix: short-commit-hash
+          base: main
+          assignees: machineFL
+          reviewers: machineAYX, jeremyliweishih, bchen1116, chukarsten, ParthivNaresh, eccabay, christopherbunn, fjlanasa, MichaelFu512, tamargrey
```

Comment:
To avoid pinging too many people, we should probably limit this, I'd recommend checking with Chris

Reply:
Hmm not sure - what does evalml typically do? Can we review with Jeremy?